src/decoder/stream.rs

```diff
@@ -2295,7 +2304,27 @@ mod tests {
         let err = reader
             .next_frame(&mut buf)
             .expect_err("Main test - expecting error");
-        assert!(matches!(err, DecodingError::Parameter(_)));
+        assert!(
+            matches!(&err, DecodingError::Parameter(_)),
+            "Unexpected kind of error: {:?}",
+            &err,
+        );
+    }
+
+    /// Tests that [`Reader.next_frame_info`] will report a `PolledAfterEndOfImage` error when
+    /// called when decoding a PNG that only contains a single frame.
+    #[test]
+    fn test_next_frame_info_polling_after_end_non_animated() {
+        let mut reader = create_reader_of_ihdr_idat();
+
+        let err = reader
+            .next_frame_info()
+            .expect_err("Main test - expecting error");
+        assert!(
+            matches!(&err, DecodingError::Parameter(_)),
+            "Unexpected kind of error: {:?}",
+            &err,
+        );
```

Comment:
This test would fail (returning `FormatError` / `MissingImageData` error instead) without taking into account the "One remaining frame will be consumed by the `finish_decoding` call below" piece at the top of the new API.